sdk/core/azure-core-http-jdk-httpclient/src/main/java/com/azure/core/http/jdk/httpclient/JdkHttpResponseSync.java

```diff
@@ -73,11 +73,15 @@ public BinaryData getBodyAsBinaryData() {
     @Override
     public void writeBodyTo(WritableByteChannel channel) throws IOException {
         if (bodyBytes != null) {
-            channel.write(ByteBuffer.wrap(bodyBytes));
+            try {
+                channel.write(ByteBuffer.wrap(bodyBytes));
+            } catch (IOException ex) {
```

Comment:
Why are we wrapping this when the API is checked with `IOException`? This seems to be wrapping needlessly.

Reply:
I think this all should be changed to have the IOException be thrown without wrapping as an UncheckedIOException as that is breaking the contract defined in the Javadocs.

```java
/**
     * Transfers body bytes to the {@link WritableByteChannel}.
     * @param channel The destination {@link WritableByteChannel}.
     * @throws IOException When I/O operation fails.
     * @throws NullPointerException When {@code channel} is null.
     */
```

Reply:
I didn't realize we never shipped stabIe jdkclient. In this case, I'm fully in favor of breaking change anywhere we already wrap  `IOException` (in methods that supposed to throw it)

Reply:
If I'm going to guess what happened is when this was initially being implemented `LOGGER.logThrowableAsError(ioException)` caused Spotbugs to incorrectly flag this as a bug (https://github.com/spotbugs/spotbugs/issues/1219) so `LOGGER.logExceptionAsError` was used wrapping with an UncheckedIOException. Instead, the logging should happen on one line and then the exception gets thrown without wrapping (and suppress our Checkstyle rule for logging exceptions [though it shouldn't raise this as a bug but I think it does]).